kernel/timeout.c

```diff
@@ -73,7 +73,8 @@ static int32_t next_timeout(void)
 {
 	struct _timeout *to = first();
 	int32_t ticks_elapsed = elapsed();
-	int32_t ret = to == NULL ? MAX_WAIT : MAX(0, to->dticks - ticks_elapsed);
+	int32_t ret = to == NULL ? MAX_WAIT :
+		MAX(0, (int32_t)(to->dticks - ticks_elapsed));
```

Comment:
ticks_elapsed is already signed, would it make any difference if the (int32_t) cast was applied directly to `to->dticks` or does it not really matter?

good catch btw

Reply:
But since #28034 was merged `to->dticks` is also a signed type.  So is this still necessary?

Reply:
Thanks @pabigot. Looks like your change would take care of it. It'd have saved me some time if I had seen the PR earlier!